packages/grid/x-data-grid-premium/src/hooks/features/rowGrouping/gridRowGroupingInterfaces.ts

```diff
@@ -40,3 +43,12 @@ export interface GridRowGroupingApi {
    */
   setRowGroupingCriteriaIndex: (groupingCriteriaField: string, groupingIndex: number) => void;
 }
+
+export interface GridGroupingRule<R = any, V = any> {
+  field: string;
+  groupingValueGetter?: (
+    params: GridGroupingValueGetterParams<V, R>,
+  ) => GridKeyValue | null | undefined;
```

Comment:
Maybe reuse definition from `GridColDefPremium`?
```suggestion
  groupingValueGetter?: GridColDefPremium<R, V>['groupingValueGetter'];
```